### ISSUE TITLE
carddav sync error:  Invalid vCard date-time string - no access to addressbook

### ISSUE THREAD

**Issue by yeupou** [the-djmaze/snappymail#528]
Hello,

**Describe the bug**
Carddav (Nextcloud) was working fine until recently, I think until 2.18. But I cannot exclude changes on Nextcloud either.

**To Reproduce**
1. No change to a working carddav config
2. Addressbook
3. "Request Failed" as a browser pop-up
4. Logs:
```
2022/09/25 11:05:33 [error] 1572#1572: *56 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Sabre\VObject\InvalidDataException: Invalid vCard date-time string: 2017-10-04T16:16:42+0000 in /usr/share/snappymail/snappymail/v/2.18.2/app/libraries/Sabre/VObject/DateTimeParser.php:315
Stack trace:
#0 /usr/share/snappymail/snappymail/v/2.18.2/app/libraries/Sabre/VObject/Property/VCard/TimeStamp.php(50): Sabre\VObject\DateTimeParser::parseVCardDateTime()
#1 /usr/share/snappymail/snappymail/v/2.18.2/app/libraries/Sabre/VObject/Property.php(302): Sabre\VObject\Property\VCard\TimeStamp->getJsonValue()
#2 /usr/share/snappymail/snappymail/v/2.18.2/app/libraries/Sabre/VObject/Component/VCard.php(463): Sabre\VObject\Property->jsonSerialize()
#3 [internal function]: Sabre\VObject\Component\VCard->jsonSerialize()
#4 /usr/share/snappymail/snappymail/v/2.18.2/app/libraries/MailSo/Base/Utils.php(836): json_encode()
#5 /usr/share/snappymail/snappymail/v/2.18.2/app/libraries/RainLoop/ServiceActions.php(193): MailSo\Base\Utils::Php2js()
```

**Expected behavior**
Access to addressbook
**Please complete the following information:**
- Browser: Firefox 104.0.2
- IMAP daemon: Dovecot
- PHP version: 7.4
- SnappyMail Version: 2.18.2

Thunderbird sync seems to work fine with the same carddav setup, so the 2017-10-04T16:16:42+0000 should probably not be a problem

Regards

**Comment by phsc84** [comment]
Just tested CardDAV sync on my installation with 2.18.3. It worked perfectly fine for me.

**Comment by yeupou** [comment]
Sorry for the delay, I was away. It is fixed on 2.18.3. Thanks a lot for your good work!
